In [17]:
import numpy as np

### 이 학습의 목적은 파이썬과 numpy 모듈의 기능을 통한 간단한 수식을 포함한 예제로 스스로 선형 회귀 학습을 실시하는 것이다.

In [18]:
# x값과 y값
x =[ 2, 4, 6, 8]
y = [81, 93, 91, 97]

In [19]:
# x와 y의 평균값 

mx = np.mean(x)
my = np.mean(y)
print("x의 평균값",mx)
print("y의 평균값",my)

x의 평균값 5.0
y의 평균값 90.5


In [20]:
# 기울기 공식의 분모
divisor = sum([(mx - i)**2 for i in x])

In [21]:
# 기울기 공식의 분자 

def top(x, mx, y, my):
    d = 0
    for i in range(len(x)):
        d += (x[i]-mx) * (y[i] -my)
    return d
dividend = top(x, mx, y, my)

In [22]:
print("분모", divisor)
print("분자:", dividend)

분모 20.0
분자: 46.0


In [23]:
# 기울기와 x 절편 구하기
a = dividend / divisor
b = my - (mx*a)

# 출력으로 확인
print("기울기 a=", a)
print('y 절편 b="', b)

기울기 a= 2.3
y 절편 b=" 79.0


## 파이썬으로 최소 제곱법을 구현해 기울기 a의 값과 y 절편 b의 값이 각각 2.3과 79임을 구할 수 있었다.